npe2/_plugin_manager.py

```diff
@@ -135,14 +135,20 @@ def iter_compatible_readers(self, paths: List[str]) -> Iterator[ReaderContributi
             return
         assert isinstance(path, str)
 
+        # use lower() to make matching case-insensitive
+        path = path.lower()
+
         if os.path.isdir(path):
             yield from (r for pattern, r in self._readers if pattern == "")
         else:
             # not sure about the set logic as it won't be lazy anymore,
             # but would we yield duplicate anymore.
             # above does not have have the unseen check either.
             # it's easy to make an iterable version if we wish, or use more-itertools.
-            yield from {r for pattern, r in self._readers if fnmatch(path, pattern)}
+            # match against pattern.lower() to make matching case insensitive
+            yield from {
+                r for pattern, r in self._readers if fnmatch(path, pattern.lower())
```

Comment:
fnmatch should be case insensitive
https://docs.python.org/3/library/fnmatch.html#fnmatch.fnmatch

It looks like problem is somewhere else.

Reply:
Only on windows? See the link:
>Both parameters are case-normalized using [os.path.normcase()](https://docs.python.org/3/library/os.path.html#os.path.normcase).

https://docs.python.org/3/library/os.path.html#os.path.normcase
which says
> On other operating systems, return the path unchanged.

Reply:
@Czaki 
Here's some tests on macOS:
```
In [9]: test_path = Path('/Users/piotrsobolewski/Downloads/IMG_2767.JPG')

In [10]: test_path.exists()
Out[10]: True

In [11]: fnmatch.fnmatch(test_path, '*.jpg')
Out[11]: False

In [12]: fnmatch.fnmatch(test_path, '*.JPG')
Out[12]: True

In [13]: fnmatch.fnmatch(test_path.lower(), '*.jpg')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[13], line 1
----> 1 fnmatch.fnmatch(test_path.lower(), '*.jpg')

AttributeError: 'PosixPath' object has no attribute 'lower'

In [14]: fnmatch.fnmatch(str(test_path).lower(), '*.jpg')
Out[14]: True

```